catroid/src/main/java/org/catrobat/catroid/merge/MergeProjectNameDialogFragment.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.merge;
+
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.Context;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.annotation.RequiresApi;
+import android.view.View;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.ui.recyclerview.dialog.TextInputDialog;
+import org.catrobat.catroid.utils.Utils;
+
+public class MergeProjectNameDialogFragment extends DialogFragment {
```

Comment:
Sorry what does this mean I do not get it.

Reply:
In agile development you split up tasks into tickets and work on those (ticket by ticket) one after another. So you should plan for each ticket what should be done, finish the ticket, and then move on to the next one. This should result in one commit per ticket and each commit should only fulfill the specified task(s): stuff which may be needed in the future (YAGNI) or isn't used at this time should NOT be in your commit.

This dialog is never called in your code, so we can't test it. 

Next thing is that i saw your commit is about fixing 3 tickets, so you may have to rethink what your tickets specifies. Backlog grooming/Ticket refactoring means you should rethink if your tickets are  specified correct. (either 3 tickets 3 commit/PR; or 1 ticket 1 commit/PR)

e.g you are working on your ticket with - the merge dialog
Since there is no actual merge feature and also no call for merge feature which should call this dialog (as far as I understand it) this ticket is actually not "Ready for development" because it depends on other features which need to be done beforehand

Only implement stuff you actually need HERE and which fulfills the description of ONE ticket

Reply:
Okay thanks!